services/core/shared/core/compat/sdk_v5/accounts.js

```diff
@@ -264,44 +270,70 @@ const indexAccountsbyPublicKey = async (accountInfoArray) => {
 
 const getLegacyAccountInfo = async ({ publicKey }) => {
 	const legacyAccountInfo = {};
-	const accountInfo = await coreApi.getLegacyAccountInfo(publicKey);
-	if (accountInfo) {
-		const legacyAddressBuffer = Buffer.from(accountInfo.address, 'hex');
-		const legacyAddress = `${legacyAddressBuffer.readBigUInt64BE().toString()}L`;
+
+	// Check if the account was already migrated
+	const reclaimTxModuleAssetId = '1000:0';
+	const transactionsDB = await getTransactionsIndex();
+	const [reclaimTx] = await transactionsDB.find({
+		senderPublicKey: publicKey,
+		moduleAssetId: reclaimTxModuleAssetId,
+	});
+
+	if (reclaimTx) {
 		Object.assign(
 			legacyAccountInfo,
 			{
-				address: getBase32AddressFromPublicKey(publicKey),
 				legacyAddress: getLegacyAddressFromPublicKey(publicKey),
-				publicKey,
-				// The account hasn't migrated/reclaimed yet
-				// So, has no outgoing transactions/registrations on the (legacy) blockchain
-				isMigrated: false,
-				isDelegate: false,
-				isMultisignature: false,
-				token: { balance: BigInt('0') },
-				legacy: {
-					...accountInfo,
-					address: legacyAddress,
-				},
+				isMigrated: true,
 			},
 		);
 	} else {
-		// Check if the account was already migrated
-		const reclaimTxModuleAssetId = '1000:0';
-		const transactionsDB = await getTransactionsIndex();
-		const [reclaimTx] = await transactionsDB.find({
-			senderPublicKey: publicKey,
-			moduleAssetId: reclaimTxModuleAssetId,
-		});
-		if (reclaimTx) {
+		let accountInfo;
+		let cachedAccountInfoStr;
+
+		const legacyHexAddress = getLegacyHexAddressFromPublicKey(publicKey);
+
+		// Fetch legacy account info from core, on timeout try querying the cache
```

Comment:
Isn't this logic should be opposite? (I mean, to avoid unnecessary calls to `lisk-core`, we can first try to fetch legacy infiormation from cache, if not exists that only call core)

Reply:
The idea is to fetch the information from the Core and use the cache as a fallback. If we use the cache first, we'd never hit the Lisk Core.

Reply:
Definitely cache first, retrieval only on cache miss.

Reply:
@MichalTuleja Sure, I'll then revert the change and then place the cache to avoid making calls to the Core.